### PR TITLE
Use None instead of -1 to fallback to default.

### PR DESCRIPTION
**Pull request by Kentzo**
Refs #59.

### CODE REVIEW

File: pycares/_cfficore/__init__.py [hunk 1]
```diff
@@ -477,15 +477,15 @@ def process_fd(self, read_fd, write_fd):
...
+    def timeout(self, t = None):
         maxtv = _ffi.NULL
         tv = _ffi.new("struct timeval*")
 
         if t >= 0.0:
             maxtv = _ffi.new("struct timeval*")
             maxtv.tv_sec = int(math.floor(t))
             maxtv.tv_usec = int(math.fmod(t, 1.0) * 1000000)
-        elif t == -1:
+        elif t in (None, -1):
```
Comment (saghul) [inline pycares/_cfficore/__init__.py]: you will need to reverse this test, because None can no longer be compared to floats on some Python versions

**Comment by Kentzo** [comment]
Perhaps `None` should mapped to `-1` / `-1.0` before reaching C-part?

File: pycares/_cfficore/__init__.py (hunk 1 above)
Comment (Kentzo) [reply pycares/_cfficore/__init__.py]: Do you recall Python's version so I could test against it?

Comment (saghul) [reply pycares/_cfficore/__init__.py]: You can check on the Travis CI output, but AFAIS Python 3.6 has this.

**Comment by saghul** [comment]
> Perhaps None should mapped to -1 / -1.0 before reaching C-part?

There is no Python part for the CPython core, it's all in C. The Python code you modified is for PyPy.

**Comment by Kentzo** [comment]
Oh ok, missed that.

File: pycares/_cfficore/__init__.py (hunk 1 above)
Comment (Kentzo) [reply pycares/_cfficore/__init__.py]: Perhaps I'm thinking of something else, but
```
>>> 1.0 == None
False
```
with 3.6.3

**Comment by saghul** [comment]
Fixed in https://github.com/saghul/pycares/pull/83
